### PR TITLE
Home controller

### PR DESCRIPTION
**Pull request by roxkon**
home-controller with basic styling for nav and footer

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+    defaults = {
+
+      site: site_name,
+      title: title,
+      icon: icon,
+      description: description,
+      keywords: %w[cannabishouse eksperyment społeczny badania naukowe cannabis house reglamentacja regla-permis nauka konopie],
+      twitter: { site_name: site_name,
+                 site: '@thecookieshq',
+                 card: 'summary',
```
Comment (wolak88) [inline app/controllers/application_controller.rb:27]: Let's change that default

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,43 @@
 class ApplicationController < ActionController::Base
+  before_action :prepare_meta_tags
+
+  protected
+
+  def prepare_meta_tags(options={})
+
+    site = "Cannabis House"
+    site_name = 'cannabishouse.eu'
+    title = [controller_name, action_name].join(" ")
```
Comment (wolak88) [inline app/controllers/application_controller.rb:10]: Shouldn't it be something like:
title = options.title || [controller_name, action_name].join(" ")

Comment (wolak88) [reply app/controllers/application_controller.rb:10]: Disregard that comment

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class PagesController < ApplicationController
+
+  def show
+    @page = Page.friendly.find_by(slug: params[:id]) if params[:id]
+    raise ActionController::RoutingError.new('Not Found') unless @page
+  	set_meta_tags(title: @page.title)
```
Comment (wolak88) [inline app/controllers/pages_controller.rb:6]: fix indentation

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
+  get ':id', to: 'pages#show', as: :page
```
Comment (wolak88) [inline config/routes.rb:2]: I think it should be at the bottom

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,9 @@
...
   get 'home/index'
   root 'home#index'
+  get 'under_construction', to: 'home#under_construction', as: :under_construction
```
Comment (wolak88) [inline config/routes.rb:5]: Why do we need `as` parameter here?

**Review by wolak88 — APPROVED** [review]
Overall LGTM 👍  left some really small comments!

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,2 +1,43 @@
...
+
+    defaults = {
+
+      site: site_name,
+      title: title,
+      icon: icon,
+      description: description,
+      keywords: %w[cannabishouse eksperyment społeczny badania naukowe cannabis house reglamentacja regla-permis nauka konopie],
+      twitter: { site_name: site_name,
+                 site: '@thecookieshq',
```
[the remark this answers is not in the document]
Comment (roxkon) [reply app/controllers/application_controller.rb:26]: do we have twitter?

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (roxkon) [reply app/controllers/pages_controller.rb:6]: wierd, i have all in place
obraz

File: config/routes.rb (hunk 4 above)
Comment (roxkon) [reply config/routes.rb:2]: git

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,2 +1,43 @@
...
+    title = [controller_name, action_name].join(" ")
+    description = "Stowarzyszenie Cannabis House"
+    icon = '/favicon.png'
+    image = '/logo-ch.jpg'
+    current_url = request.url
+
+    # Let's prepare a nice set of defaults
+
+    defaults = {
+
```
[the remark this answers is not in the document]
Comment (roxkon) [reply app/controllers/application_controller.rb:19]: what do You mean? Should we remove all defaults?

**Pull request merged**